exporter/elasticsearchexporter/integrationtest/collector.go

```diff
@@ -0,0 +1,207 @@
+		BuildInfo: component.BuildInfo{
+			Command: "otelcol",
+			Version: "v0.0.0",
+		},
+	}
+	collector, err := otelcol.NewCollector(collectorSettings)
+	require.NoError(t, err)
+
+	wrappedCollector := &otelCol{
+		col:      collector,
+		settings: collectorSettings,
+	}
+	t.Cleanup(wrappedCollector.Shutdown)
+	return wrappedCollector
+}
+
+// Recreate recreates the collector with the same configuration. Note
+// that after recreating the collector `Run` should be called to get the
+// collector running again.
+func (otel *otelCol) Recreate() error {
+	otel.mu.Lock()
+	defer otel.mu.Unlock()
+
+	if otel.col != nil {
+		otel.col.Shutdown()
+	}
+	newCollector, err := otelcol.NewCollector(otel.settings)
+	if err != nil {
+		return fmt.Errorf("failed to restart collector: %w", err)
+	}
+	otel.col = newCollector
+	return nil
+}
+
+// Run wraps the original collector run to protect the updates on the
+// collector variable.
+func (otel *otelCol) Run(ctx context.Context) error {
+	otel.mu.Lock()
+	collector := otel.col
+	otel.mu.Unlock()
+
+	if collector == nil {
+		return fmt.Errorf("nil collector")
+	}
+
+	return collector.Run(ctx)
+}
+
+// IsRunning checks if the otel collector is currently running or not.
+func (otel *otelCol) IsRunning() bool {
+	otel.mu.Lock()
+	defer otel.mu.Unlock()
+
+	if otel.col == nil {
+		return false
+	}
+	return otel.col.GetState() == otelcol.StateRunning
+}
+
+// Shutdown shuts down the otel collector. This test wrapper allows for
+// restarting the collector after shutdown by creating a new otel collector
+// with the same configuration as the original collector.
+func (otel *otelCol) Shutdown() {
+	otel.mu.Lock()
+	defer otel.mu.Unlock()
+
+	if otel.col != nil {
+		otel.col.Shutdown()
+		otel.col = nil
+	}
+}
+
+const cfgTemplate = `
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: {{.GRPCEndpoint}}
+
+exporters:
+  debug:
+    verbosity: {{if .Debug}} detailed {{else}} basic {{end}}
+  elasticsearch:
+    endpoints: [ {{.ESEndpoint}} ]
+    logs_index: {{.ESLogsIndex}}
+    sending_queue:
+      enabled: true
+      storage: file_storage/elasticsearchexporter
+      num_consumers: 1
+      queue_size: 10000000
+    flush:
+      interval: {{.FlushInterval}}
+    retry:
+      enabled: true
+      max_requests: 10000
+
+extensions:
+  file_storage/elasticsearchexporter:
+    directory: {{.StorageDir}}
+
+service:
+  extensions: [file_storage/elasticsearchexporter]
+  pipelines:
+    logs:
+      receivers: [otlp]
+      processors: []
+      exporters: [elasticsearch, debug]
+`
+
+func getCollectorCfg(cfg config, esURL, tmpDir string, flushIvl time.Duration) (*os.File, error) {
+	cfgFile, err := os.CreateTemp(tmpDir, "otelconf-*")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OTEL configuration file: %w", err)
+	}
+	tmpl, err := template.New("otel-config").Parse(cfgTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OTEL collector configuration: %w", err)
```

Comment:
nit: clearer error message
```suggestion
		return nil, fmt.Errorf("failed to create OTEL collector configuration template: %w", err)
```